src/EditDialog.cpp

```diff
@@ -194,70 +297,79 @@ void EditDialog::setNull()
+        }
     }
 }
 
 // Called when the user manually changes the "Mode" drop down combobox
-void EditDialog::editModeChanged(int editMode)
+void EditDialog::editModeChanged(int newMode)
 {
-    // If the new editor mode is "text editor", load the data into it
-    if (editMode == 0) {
-        QByteArray data = hexEdit->data();
-        QString textData = QString::fromUtf8(data, data.size());
-        ui->editorText->setPlainText(textData);
-    }
-
-    // Switch to the selected editor
-    ui->editorStack->setCurrentIndex(editMode);
+    // * If the dataSource is the text buffer, the data is always text *
+    if (dataSource == TextBuffer) {
+        switch (newMode) {
+        case TextEditor: // Switching to the text editor
+            // Nothing to do, as the text is already in the text buffer
+            break;
+
+        case HexEditor: // Switching to the hex editor
+            // Convert the text widget buffer for the hex widget
+            hexEdit->setData(ui->editorText->toPlainText().toUtf8());
+
+            // The hex widget buffer is now the main data source
+            dataSource = HexBuffer;
+            break;
+
+        case ImageViewer:
+            // Clear any image from the image viewing widget
+            ui->editorImage->setPixmap(QPixmap(0,0));
+            break;
+        }
 
-    // Show or hide the warning about editing binary data in text mode
-    int dataType = checkDataType();
-    updateBinaryEditWarning(editMode, dataType);
-}
+        // Switch to the selected editor
+        ui->editorStack->setCurrentIndex(newMode);
+        return;
+    }
 
-void EditDialog::editTextChanged()
-{
-    if(ui->editorText->hasFocus())
-    {
-        hexEdit->blockSignals(true);
-        hexEdit->setData(ui->editorText->toPlainText().toUtf8());
-        hexEdit->blockSignals(false);
+    // * If the dataSource is the hex buffer, the contents could be anything
+    //   so we just pass it to our loadData() function to handle *
+    if (dataSource == HexBuffer) {
+        // Switch to the selected editor first, as loadData() relies on it
+        // being current
+        ui->editorStack->setCurrentIndex(newMode);
 
-        // Update the cell data info in the bottom left of the Edit Cell
-        int dataType = checkDataType();
-        updateCellInfo(dataType);
+        // Load the data into the appropriate widget, as done by loadData()
+        loadData(hexEdit->data());
     }
 }
 
-void EditDialog::hexDataChanged()
-{
-    // Update the text editor accordingly
-    ui->editorText->blockSignals(true);
-    ui->editorText->setPlainText(QString::fromUtf8(hexEdit->data().constData(), hexEdit->data().size()));
-    ui->editorText->blockSignals(false);
-}
-
 // Determine the type of data in the cell
-int EditDialog::checkDataType()
+int EditDialog::checkDataType(const QByteArray& data)
 {
-    QByteArray cellData = hexEdit->data();
+    QByteArray cellData = data;
```

Comment:
Why would you create that variable? ```const QByteArray& data``` is passed, you do no non-const operations with it.
Redundant copy initialization.